lecture_3/03_project/calculator/internal/engine/handler/betRepository.go

```diff
@@ -0,0 +1,15 @@
+package handler
+
+import (
+	storagemodels "code-cadets-2021/lecture_3/03_project/calculator/internal/infrastructure/sqlite/models"
+	"context"
+
+	domainmodels "code-cadets-2021/lecture_3/03_project/calculator/internal/domain/models"
+)
+
+type BetRepository interface {
+	InsertBet(ctx context.Context, bet domainmodels.BetCalculated) error
+	UpdateBet(ctx context.Context, bet domainmodels.BetCalculated) error
+	GetBySelectionID(ctx context.Context, id string) ([]domainmodels.BetCalculated, bool, error)
```

Comment:
There is no need to return `exists` flag, if there are no results you can just return an empty slice. Note that we cannot do the same with methods where only one value is being returned, because some kind of struct is guaranteed to be returned and you cannot tell whether the empty struct as a result was returned because it didn't exist in the repository or that's the actual value which was fetched.